docs/articles/documentation/recipes/best-practices/speed-up-test-execution.md

```diff
@@ -0,0 +1,115 @@
+redirect_from:
+  - /documentation/recipes/speed-up-test-execution.html
+---
+
+# Speed Up Test Execution
+
+This section describes ways to decrease the execution time of TestCafe tests. It includes the following sections:
+
+* [Run Tests Concurrently](#run-tests-concurrently)
+* [Run Tests in Headless Browsers](#run-tests-in-headless-browsers)
+* [Use Roles for Login](#use-roles-for-login)
+* [Set Test Speed](#set-test-speed)
+* [Run Tests in Local Browsers](#run-tests-in-local-browsers)
+* [Mock Requests](#mock-requests)
+* [Optimize Your Page Model](#optimize-your-page-model)
+
+## Run Tests Concurrently
+
+Enable concurrent mode to run multiple browser instances simultaneously. This speeds up the testing process at the expense of higher resource consumption.
+
+You can enable concurrency with the [concurrency](../../reference/configuration-file.md#concurrency) configuration file property:
+
+```json
+{
+  "concurrency": 3
+}
+```
+
+In CLI, use the [--concurency](../../reference/command-line-interface.md#-c-n---concurrency-n) option:
+
+```sh
+testcafe --concurency 3 chrome tests/
+```
+
+For more info on concurrency, read [Run Tests Concurrently](../../guides/basic-guides/run-tests.md#run-tests-concurrently).
+
+## Run Tests in Headless Browsers
+
+Headless browsers take less time to initialize because they don't need to render the application. They also enable you to run tests in environments without graphical capabilities, such as CI containers.
+
+For more information, see [Test in Headless Mode](../../guides/concepts/browsers.md#test-in-headless-mode)
+
+## Use Roles for Login
+
+Use [Roles](../../guides/advanced-guides/authentication.md#user-roles) to save time on user authentication.
+
+Roles are user log-in routines. When you first enable a `Role`, TestCafe executes the routine and saves the user authentication data that it receives. When you activate the same role later, TestCafe loads the authentication data and skips the log-in process.
+
+When you switch between `Roles`, TestCafe replaces your browser's existing authentication data with the new role's credentials.
+
+Place your `Role` statements inside a [beforeEach hook](../../reference/test-api/fixture/beforeeach.md) to recall the authentication data before each of the tests in your fixture.
+
+> Roles can access authentication data in cookie and browser storage only. If your authentication system stores data elsewhere, roles may not work.
+
+For more info on Roles, see [User Roles](../../guides/advanced-guides/authentication.md#user-roles)
+
+## Set Test Speed
+
+TestCafe emulates real user actions on tested webpages. Set the [speed](../../reference/command-line-interface.md#--speed-factor) option to change the emulation speed.
+
+The highest value of `1` represents the fastest actions on the page. This is a default value. The minimum value of `0.01` represents a speed that is 100 lower. Setting a lower `speed` can be useful for debugging tests, but slows tests down.
```

Comment:
*the* default value, because there's only one.

```suggestion
The highest value of `1` represents the fastest possible action speed. This is the default `speed` value. The minimum value of `0.01` represents a speed that is 100 times lower. Setting a lower `speed` can be useful for debugging, but it slows tests down.
```